lib/npm.js

```diff
@@ -376,21 +428,23 @@ NPMLocation.prototype = {
         if (npmRes.headers['content-length'] > 50000000)
           return reject('Response too large.');
 
+        var chunks = [];
+
         npmRes.pause();
 
-        var gzip = zlib.createGunzip();
-
-        npmRes
-        .pipe(gzip)
-        .pipe(tar.Extract({
-          path: outDir,
-          strip: 1,
-          filter: function() {
-            return !this.type.match(/^.*Link$/);
-          }
-        }))
-        .on('error', reject)
-        .on('end', resolve);
+        npmRes.on('data', function (chunk) {
+          chunks.push(chunk);
+        });
+
+        npmRes.on('error', reject);
+
+        npmRes.on('end', function () {
+          Promise.resolve(Buffer.concat(chunks))
+            .then(gunzip)
+            .then(extractTarBuffer)
+            .then(resolve)
+            .catch(reject);
```

Comment:
This seems counter to the streams philosophy to inject promises.

Is there not an approach that works directly on the streams?